internals-js/src/specs/sourceSpec.ts

```diff
@@ -140,8 +144,271 @@ export class SourceSpecDefinition extends FeatureDefinition {
   sourceFieldDirective(schema: Schema) {
     return this.directive<SourceFieldDirectiveArgs>(schema, 'sourceField')!;
   }
+
+  override validateSubgraphSchema(schema: Schema): GraphQLError[] {
```

Comment:
Make multiple functions out of this one.

Reply:
https://github.com/apollographql/federation/pull/2910/commits/746fc4d5d16a50ab6486dec80cc0631921d09f12